src/main/resources/checkstyle-config.xml

```diff
@@ -30,10 +30,15 @@
 
     <module name="NeedBraces"/>
 
+    <module name="IllegalCatch">
```

Comment:
Do we want to do this? Why are all these forbidden? `Throwable` was needed by @Ben-Zvi in his very first Drill bug fix. `RuntimeException` is the root of the unchecked exceptions beloved by Drill code and is used as a general "something went wrong" placeholder.

These are all commented out, but still...

Finally, `IndexOutOfBoundsException` is a perfectly fine exception. Adding this line is probably handy to catch all reference as part of this fix, but probably not helpful as a general policy.

Reply:
I think IOBE should be caught and converted to a UserException so that the end user does not get an incomprehensible error message.

Reply:
My point is that there should be no special handling for IOBE compared to NPE or CCE. If it is in process call to a Java library all run-time exceptions including NPE, IOBE and CCE should be propagated to the caller. If it is a remote call, runtime and checked exceptions need to be handled by propagating them to a remote caller as well.

What an end user tool displays to an end user depends on a tool. IMO, the best option is to have a configuration parameter that determines verbosity level. With default verbosity level, the tool may show a generic error message (possibly explaining how to increase verbosity level to get more info, see `maven -X`), and there should be an option to change the verbosity level and print a stack trace that initially triggered the error. Such stack trace is required when an end user logs a bug or asks for help on the dev list.